clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -548,7 +551,9 @@ public void handle(JoinGroupResponse joinResponse, RequestFuture<ByteBuffer> fut
                     || error == Errors.INVALID_SESSION_TIMEOUT
                     || error == Errors.INVALID_GROUP_ID
                     || error == Errors.GROUP_AUTHORIZATION_FAILED
-                    || error == Errors.GROUP_MAX_SIZE_REACHED) {
+                    || error == Errors.GROUP_MAX_SIZE_REACHED
+                    || error == Errors.MEMBER_ID_MISMATCH) {
```

Comment:
It should be ok since the error message clearly states:
`The group.instance.id is already in the consumer, however the corresponding member.id is not matching the record on coordinator`